### ISSUE TITLE
Assertion failed: Capacity() - Size() > 0, file C:\buildbot\slave\builds\engine-win32-64-master\build\tmp\dynamo_home\sdk\include\dmsdk/dlib/array.h, line 464

### ISSUE THREAD

**Issue by subsoap** [defold/editor2-issues#2468]
![counts](https://user-images.githubusercontent.com/409170/52624078-c21f1380-2e62-11e9-8794-7fbfe4304a19.png)

Probably related to this, trying to make A LOT OF STUFF HAPPEN AT ONCE.

**Comment by subsoap** [comment]
Looks like I'm just not allowed to increase the Max Particlefx Count above 64?

```
ERROR:GUI: Particle FX gui component buffer is full (128), component disregarded. Increase 'gui.max_particlefx_count' as needed
Assertion failed: Capacity() - Size() > 0, file C:\buildbot\slave\builds\engine-win32-64-master\build\tmp\dynamo_home\sdk\include\dmsdk/dlib/array.h, line 464
```

**Comment by erikangelin-king** [comment]
#2464

**Comment by subsoap** [comment]
Yeah, I'm getting ```Assertion failed: Capacity() - Size() > 0, file C:\buildbot\slave\builds\engine-win32-64-master\build\tmp\dynamo_home\sdk\include\dmsdk/dlib/array.h, line 464``` constantly with this, might be related to GUI particlefx somehow.

**Comment by subsoap** [comment]
Now I reset all of the values back to default for the maxes and it's still happening. 😢

**Comment by subsoap** [comment]
[_crash (2).zip](https://github.com/defold/editor2-issues/files/2854998/_crash.2.zip)

Here's a new crash dump I noticed I have but it's older than these crashes.

**Comment by subsoap** [comment]
@mathiaswking I can get this to happen consistently with this project now. Can you remind me what e-mail address to send it to?

**Comment by danengelbrecht-king** [comment]
So, you are using a lot of particles in a GUI, or just a lot of particles in a regular GO collection?
You have settings of for particles both under the gui section and under the particle_fx section.
Anyway, we should not assert, we should log an error and not crash.

Could you either send it as a private message on the forum and attach the project (I'm "vilse" on the forum) or you can use slack to send me the project there, there my username is "Dan Engelbrecht") and I'll have a look.

**Comment by subsoap** [comment]
I sent @mathiaswking a small reproduction project zip. I'll send you one privately on there too.

**Comment by danengelbrecht-king** [comment]
Created jira DEF-3774

**Comment by AGulev** [comment]
Issue was fixed in 1.2.147 https://forum.defold.com/t/defold-1-2-147-has-been-released/43851
